modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java

```diff
@@ -0,0 +1,610 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.swap.type;
+
+
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.PayReceive;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.BusinessDayAdjustment;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.basics.index.PriceIndex;
+import com.opengamma.strata.basics.schedule.Frequency;
+import com.opengamma.strata.basics.schedule.PeriodicSchedule;
+import com.opengamma.strata.product.swap.InflationRateCalculation;
+import com.opengamma.strata.product.swap.NotionalSchedule;
+import com.opengamma.strata.product.swap.PaymentSchedule;
+import com.opengamma.strata.product.swap.RateCalculationSwapLeg;
+
+/**
+ * A market convention for the floating leg of rate swap trades based on a price index.
+ * <p>
+ * This defines the market convention for a floating leg based on the observed value
+ * of a Price index such as 'GB_HICP' or 'US_CPI_U'.
+ */
+@BeanDefinition
+public final class InflationRateSwapLegConvention
+    implements SwapLegConvention, ImmutableBean, Serializable {
+
+  /**
+   * The Price index.
+   * <p>
+   * The floating rate to be paid is based on this price index
+   * It will be a well known price index such as 'GB_HICP'.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final PriceIndex index;
+
+  /**
+   * The leg currency, optional with defaulting getter.
```

Comment:
I don't think "optional with defaulting getter" adds anything. I think the Javadoc would be clearer without it.